### ISSUE TITLE
Stop processing the channel queue once the channel is closed

### ISSUE THREAD

**Issue by hackaugusto** [raiden-network/raiden#1611]
A channel can be closed while there are still messages queued. The queue **for the closed channel** must be stopped.

Follow up for: https://github.com/raiden-network/raiden/issues/454

**Comment by LefterisJP** [comment]
@andrevmatos What's the status here?

**Comment by andrevmatos** [comment]
I'm working on it, kind of depending on #2208, but already fixed that one, should be merged soon
Will finish it after merging that and finishing new contracts migration

**Comment by hackaugusto** [comment]
> a namedtuple(chain_id: ChainID, token_network_id: TokenNetworkID, channel_id: ChannelID)

Please, don't use a namedtuple, these are not the best for the serialization. For now I believe that a plain class is good enough.

Additionally, this may not be required ATM, although it's a nice to have change.

**Comment by andrevmatos** [comment]
@hackaugusto I don't agree with you, namedtuple is a simple tuple, can be [de]serialized even for a json array, and `typing.NamedTuple` is declared exactly like a simple/plain class. Plain classes will require lots of boilerplate, only thing better for serialization would be `attrs`/`dataclasses`, but I don't think we need to introduce it now (I'd like to push for it after Red Eyes).
About second comment, the first point is not required, but is less work than adding all the parameters required without it in the second point, so I'd push for it as I'd do it anyway, and will make it less error prone.

**Comment by hackaugusto** [comment]
Instances where namedtuples where not good enough:

2e5d63cb5d60344438083a5293f7674a53bb59b6 : refactored a namedtuple to a class to work with our serialization code
592f920ec14d86b1ed34f8489025129274bb0b09 : refactored a namedtuple to a typing.NamedTuple for type annotations
5ebe70fe7a073a071a914a4a6936c113df5396e2 : refactored a namedtuple because for pickle serialization it doesn't work with nested classes

I'm not saying it won't work for what you're doing right now, but I'm asking you to not use it to avoid problems down the line.

**Comment by andrevmatos** [comment]
`typing.NamedTuple` was the initial plan, just used namedtuple as a generic reference to it, as the former is a subtype of the later. For pickle, it should be droped soon, and in this case, it won't contain nested classes, only constant `bytes`/`int`

**Comment by LefterisJP** [comment]
As discussed in the daily this is not making it for this release. Pushing it to next milestone.
